### ISSUE TITLE
certbot renew --debug-challenges

### ISSUE THREAD

**Issue by wolfgang42** [certbot/certbot#5005]
I'm using `certbot` on Ubuntu 16.04, installed from `ppa:certbot/certbot`. My server is failing to renew the certificates; I have a suspicion that this has to do with where the HTTP daemon is looking for the `/.well-known/acme-challenge/` directory and am attempting to debug. I found the `--debug-challenge` flag, and tried to use it:
```
$ sudo certbot renew --no-self-upgrade --cert-name linestarve.com --debug-challenge
Saving debug log to /var/log/letsencrypt/letsencrypt.log

-------------------------------------------------------------------------------
Processing /etc/letsencrypt/renewal/linestarve.com.conf
-------------------------------------------------------------------------------
Cert is due for renewal, auto-renewing...
Renewing an existing certificate
Performing the following challenges:
http-01 challenge for linestarve.com
Waiting for verification...

-------------------------------------------------------------------------------
Challenges loaded. Press continue to submit to CA. Pass "-v" for more info about
challenges.
-------------------------------------------------------------------------------
```
This looks promising, and I began to debug the problem. Unfortunately, a few seconds later it forged ahead, heedless of my attempts to halt it:
```
Cleaning up challenges
Attempting to renew cert from /etc/letsencrypt/renewal/linestarve.com.conf produced an unexpected error: Failed authorization procedure. linestarve.com (http-01): urn:acme:error:unauthorized :: The client lacks sufficient authorization :: Invalid response from http://linestarve.com/.well-known/acme-challenge/PkEEMAWYYIhIfo8awvYBOKXZqnkIOlDruIYUgBDttEk:
(and so on)
```

Some investigating indicates that the `notify(..., pause=True)` argument is ignored in noninteractive mode (which `renew` runs in), so I tried `--force-interactive`. Unfortunately that results in "`--force-interactive cannot be used with renew`".

Is there a way to use `--debug-challenges` while renewing a certificate?

**Comment by SwartzCr** [comment]
This is a really great question - @bmw any idea? This seems like an essential feature

**Comment by bmw** [comment]
`--debug-challenges` currently cannot be used with `certbot renew`. The reason for this is `certbot renew` is usually run automatically from something like `cron` and we didn't want Certbot to prompt the user and hang indefinitely. With that said, this feature would be valuable and we could potentially make it work with `renew` by including `--force-interactive` on the command line which was suggested in #4925.

In the meantime, you should be able to use `--debug-challenges` by running the following command instead:
```
certbot certonly --cert-name linestarve.com --debug-challenges
```

**Comment by wolfgang42** [comment]
I guess there are two different changes to be made:
* `certbot renew` should allow `--force-interactive`. This is covered by #4925.
* `--debug-challenges` should return an error if running in noninteractive mode, rather than showing the "press continue" message and then carrying on regardless. (I hit the rate-limit on failed renewals at one point because I kept trying different flags to get it to stop and it kept ignoring them and requesting renewals anyway.)

I managed to fix the problem already, but thank you for your workaround. 😃

**Comment by francisrod01** [comment]
The same problem where:

```
Cleaning up challenges
Failed authorization procedure.my-website.domain.com (http-01): urn:acme:error:unauthorized :: The client lacks sufficient authorization :: Invalid response from http://my-website.domain.com/.well-known/acme-challenge/<something-here>: "<!DOCTYPE html>
<
```

**Comment by SwartzCr** [comment]
@francisrod01 just to be clear, are you also using the `--debug-challenges` flag? or just getting a similar error?

**Comment by francisrod01** [comment]
In fact, I needed to allow `.well-known/` in nginx first.

**Comment by wumpus** [comment]
It'd be nice if at least this issue and workaround could be documented. Right now the documentation says to do something that does not ever work.

**Comment by bmw** [comment]
I agree! Would someone like to submit a PR?

The file to modify is `certbot/cli.py`. In the line where `--debug-challenges` flag is added, you can remove `renew` from the list so the flag isn't shown with `certbot --help renew`. Adding a sentence to the help text that the flag doesn't work with the renew subcommand can be done in addition if you want.

**Comment by r5d** [comment]
@bmw, I've opened a [PR](https://github.com/certbot/certbot/pull/6141).

**Comment by bmw** [comment]
Fixed in #6141.
